_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,522 @@
+
+Some library maintainers have long avoided case classes in public APIs, since any change to their fields affected binary compatibility. This was typically observed whenever a new field was added to the case class primary constructor, even if it was defined with a default value to resolve source compatibility problems. Though such binary incompatibilities were detected by [MiMa](https://github.com/lightbend/mima), avoiding them required manually creating shim methods to match the old signatures.
+
+The `@unroll` annotation will eliminate this long-standing issue. This annotation lets you add additional parameters to method `def`s, `class` constructors, or `case class`es, without breaking binary compatibility. `@unroll` works by generating unrolled or “telescoping” forwarders, ensuring your code will remain binary compatible as new default parameters and fields are added over time.
+
+```scala
+//> using scala 3.7
+//> using options -experimental
+
+import scala.annotation.unroll
+
+case class Product(
+    name: String,
+    price: BigDecimal,
+    @unroll description: Option[String] = None,
+){
+  // generated by compiler:
+  // def this(name: String, price: BigDecimal) = this(name, price, default$1)
+  // def copy(name: String, price: BigDecimal) = this(name, price)
+}
+
+object Factory:
+  def processOrder(
+      products: List[Product],
+      @unroll quantity: Int = 1,
+      @unroll discount: Int = 0
+  ): Unit = ???
+
+  // generated by the compiler:
+  // def processOrder(products: List[Product]): Unit = processOrder(products, default$1, default$2)
+  // def processOrder(products: List[Product], quantity: Int): Unit = processOrder(products, quantity, default$2)
+```
+
+More details can be found in [`@unroll` reference](https://docs.scala-lang.org/scala3/reference/experimental/unrolled-defs.html)
+
+### [SIP-68: Reference-able Package Objects](https://github.com/scala/improvement-proposals/pull/100)
+
+One limitation with `package object`s is that we cannot directly refer to them as values, even though that is possible for normal `object`s. Such references can already be made with a special ".`package`" accessor, but this is ugly and non-obvious. Or one could use a normal `object`, which is not always possible.
+
+The SIP drops this limitation, allowing such direct references.
+
+```scala
+//> using scala 3.7
+//> using options -experimental
+
+package my.pkg
+import scala.language.experimental.packageObjectValues
+
+package object releases:
+  val `3.7` = "3.7.0" 
+  
+@main def referencablePkgObject = 
+  val versions = my.pkg.releases
+  val oldStyle = my.pkg.releases.`package`
+  println(versions eq oldStyle) // prints true
+```
+
+For more details visit the [reference page](https://docs.scala-lang.org/scala3/reference/experimental/package-object-values.html) for this feature.
+
+## Other notable changes
+
+### New scheme for given prioritization
```

Comment:
As this is kind-of breaking change, shouldn't we somehow point this out at the very beginning of the blog post?